pyvrp/Model.py

```diff
@@ -0,0 +1,291 @@
+from typing import List, Optional, Union
+from warnings import warn
+
+import numpy as np
+
+from pyvrp.GeneticAlgorithm import GeneticAlgorithm, GeneticAlgorithmParams
+from pyvrp.PenaltyManager import PenaltyManager
+from pyvrp.Population import Population, PopulationParams
+from pyvrp.Result import Result
+from pyvrp._Individual import Individual
+from pyvrp._ProblemData import Client, ProblemData
+from pyvrp._XorShift128 import XorShift128
+from pyvrp.crossover import selective_route_exchange as srex
+from pyvrp.diversity import broken_pairs_distance as bpd
+from pyvrp.exceptions import ScalingWarning
+from pyvrp.stop import StoppingCriterion
+
+Depot = Client
+
+
+class Edge:
+
+    __slots__ = ["frm", "to", "distance", "duration"]
+
+    def __init__(self, frm: Client, to: Client, distance: int, duration: int):
+        self.frm = frm
+        self.to = to
+        self.distance = distance
+        self.duration = duration
+
+
+class VehicleType:
+
+    __slots__ = ["number", "capacity"]
+
+    def __init__(self, number: int, capacity: int):
+        self.number = number
```

Comment:
We use `qty` nowhere else in the code, but we do talk about number/num a lot. E.g. `numVehicles`, `numClients`, `numRoutes`. Something like `numAvail` could work? Or perhaps just `numVehicles`. But that might be confusing since it could also mean the overall number of vehicles across all types.

Reply:
I want to finish this PR so I'll merge it now with `number`, but let's revisit this for heterogeneous capacity. I think something like `numAvail`/`numVehicles` could work.